File: app/src/main/java/com/jgm90/cloudmusic/adapters/SlidePagerAdapter.java
package com.jgm90.cloudmusic.adapters;

import android.content.Context;
import android.text.TextUtils;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.RelativeLayout;
import android.widget.TextView;

import androidx.annotation.NonNull;
import androidx.appcompat.widget.AppCompatImageView;
import androidx.viewpager.widget.PagerAdapter;

import com.jgm90.cloudmusic.GlideApp;
import com.jgm90.cloudmusic.R;
import com.jgm90.cloudmusic.models.SongModel;
import com.jgm90.cloudmusic.utils.SharedUtils;

import java.util.List;

public class SlidePagerAdapter extends PagerAdapter {

    private Context context;
    private LayoutInflater inflater;
    private List<SongModel> items;

    private SongModel song;

    public SlidePagerAdapter(Context context, List<SongModel> items) {
        this.context = context;
        this.items = items;
    }

    @Override
    public int getCount() {
        return items.size();
    }

    @Override
    public boolean isViewFromObject(@NonNull View view, @NonNull Object object) {
        return (view == object);
    }

    @NonNull
    @Override
    public Object instantiateItem(@NonNull ViewGroup container, int position) {
        inflater = (LayoutInflater) context.getSystemService(Context.LAYOUT_INFLATER_SERVICE);
        View view = inflater.inflate(R.layout.slide, container, false);
        AppCompatImageView albumArt = view.findViewById(R.id.album_art);
        RelativeLayout lyricsView = view.findViewById(R.id.lyrics_view);
        TextView currentLine = view.findViewById(R.id.lbl_currect_line);
        TextView nextLine = view.findViewById(R.id.lbl_next_line);
        albumArt.setImageResource(R.drawable.default_cover);
        String picUrl;
        song = items.get(position);
        view.setTag("rootView_" + song.getId());
        lyricsView.setTag("lyricsView_" + song.getId());
        currentLine.setTag("currentView_" + song.getId());
        nextLine.setTag("nextView_" + song.getId());
        if (!TextUtils.isEmpty(song.getLocal_thumbnail())) {
            picUrl = song.getLocal_thumbnail();
        } else {
            picUrl = SharedUtils.server + "pic/" + song.getPic_id();
        }
        GlideApp
                .with(context)
                .load(picUrl)
                .centerCrop()
                .placeholder(R.drawable.default_cover)
                .into(albumArt);
        container.addView(view);
        return view;
    }

    @Override
    public void destroyItem(@NonNull ViewGroup container, int position, @NonNull Object object) {
        container.removeView((RelativeLayout) object);
    }
}

File: app/src/main/java/com/jgm90/cloudmusic/adapters/SlidePagerAdapter.kt
package com.jgm90.cloudmusic.adapters

import android.content.Context
import android.text.TextUtils
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.RelativeLayout
import android.widget.TextView
import androidx.appcompat.widget.AppCompatImageView
import androidx.viewpager.widget.PagerAdapter
import com.jgm90.cloudmusic.GlideApp
import com.jgm90.cloudmusic.R
import com.jgm90.cloudmusic.models.SongModel
import com.jgm90.cloudmusic.utils.SharedUtils

class SlidePagerAdapter(
    private val context: Context,
    private val items: List<SongModel>,
) : PagerAdapter() {
    private var song: SongModel? = null

    override fun getCount(): Int = items.size

    override fun isViewFromObject(view: View, `object`: Any): Boolean = view === `object`

    override fun instantiateItem(container: ViewGroup, position: Int): Any {
        val inflater = context.getSystemService(Context.LAYOUT_INFLATER_SERVICE) as LayoutInflater
        val view = inflater.inflate(R.layout.slide, container, false)
        val albumArt: AppCompatImageView = view.findViewById(R.id.album_art)
        val lyricsView: RelativeLayout = view.findViewById(R.id.lyrics_view)
        val currentLine: TextView = view.findViewById(R.id.lbl_currect_line)
        val nextLine: TextView = view.findViewById(R.id.lbl_next_line)
        albumArt.setImageResource(R.drawable.default_cover)
        song = items[position]
        val currentSong = song ?: return view
        view.tag = "rootView_${currentSong.id}"
        lyricsView.tag = "lyricsView_${currentSong.id}"
        currentLine.tag = "currentView_${currentSong.id}"
        nextLine.tag = "nextView_${currentSong.id}"
        val picUrl = if (!TextUtils.isEmpty(currentSong.local_thumbnail)) {
            currentSong.local_thumbnail
        } else {
            SharedUtils.server + "pic/" + currentSong.pic_id
        }
        GlideApp
            .with(context)
            .load(picUrl)
            .centerCrop()
            .placeholder(R.drawable.default_cover)
            .into(albumArt)
        container.addView(view)
        return view
    }

    override fun destroyItem(container: ViewGroup, position: Int, `object`: Any) {
        container.removeView(`object` as RelativeLayout)
    }
}
